generate-sales-report/src/Code.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Default values
+var ACCOUNT_NAME = 'Acme';
+var REGION = 'Midwest';
+
+/**
+ * Creates a slide deck of sales opportunity data from the spreadsheet.
+ */
+function generateReport() {
+  var spreadsheet = SpreadsheetApp.getActive();
+  var sheet = spreadsheet.getSheetByName('Data Results');
+  var dataRange = sheet.getDataRange();
+  var numRows = dataRange.getNumRows();
+  ACCOUNT_NAME = spreadsheet.getRangeByName('AccountName').getValue();
+  REGION = spreadsheet.getRangeByName('Region').getValue();
+  
+  var presentationID = copyReportTemplate();
+  var presentation = SlidesApp.openById(presentationID);
+  
+  var slides = presentation.getSlides();
+  introSlide(slides[0], presentation);
+  leadsSlide(slides[1], sheet, presentation, numRows);
+  
+  // Create a charts sheet to delete later
+  var tempSheet = spreadsheet.insertSheet('Charts Sheet', spreadsheet.getSheets().length);
+  stageSlide(slides[2], presentation, sheet, tempSheet, numRows);
+  businessSlide(slides[3], presentation, sheet, tempSheet, numRows);
+  topSlide(slides[4], sheet);
+  nearSlide(slides[5], sheet);
+  needsSlide(slides[6], sheet);
+
+  spreadsheet.deleteSheet(tempSheet);
+}
+
+function introSlide(slide, presentation) {
```

Comment:
can you either add function comments OR make them private functions by appending '_' to the name:
function introSlide_(.....

https://developers.google.com/apps-script/guides/html/communication#private_functions

Reply:
all functions except generateReport() are now private